src/main/vcp_hal/usbd_cdc_interface.c

```diff
@@ -187,6 +195,12 @@ static int8_t CDC_Itf_Control (uint8_t cmd, uint8_t* pbuf, uint16_t length)
     LineCoding.format     = pbuf[4];
     LineCoding.paritytype = pbuf[5];
     LineCoding.datatype   = pbuf[6];
+    // If a callback is provided, tell the upper driver of changes in baud rate
```

Comment:
The structure is just build here - just test that `length == 7` (pbuf[6] is last byte)

Reply:
Yuk! No, it should be the sizeof the structure and the use of the packed attribute facilitates this.

Reply:
`USBD_CDC_LineCodingTypeDef LineCoding` is used here, that structure is actually not packed ... 
And the structure is rebuilt  byte-by-byte, it's size and structure is defined by unpack code above, not `struct` declaration ...
When casting received packet to the structure it will make sense, but alignment and endianness is problem then

Also note that `length` is actually 8 here(if the code is working, IMO USB is using 16bit units a lot ...

Reply:
Actually it seems that this code path is simply untested ...